### PR TITLE
Docker tests

### PR DESCRIPTION
**Pull request by mbologna**
Inspired by issue #108, I pushed two Docker container images:
1. salt-master and salt-api
2. salt-minion

Every container accepts a volume in which you specify the configuration along with certs and keys.
In this way, the container image is general and you launch it just by specifying a different configuration.
Everything is put under docker/ directory.

I also changed travis build to use launch these containers during tests, and I added a small JUnit test to make sure that this is all working correctly.

### CODE REVIEW

File: docker/etc_minion1/salt/minion [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+
+# Per default the minion will automatically include all config files
+# from minion.d/*.conf (minion.d is a directory in the same directory
+# as the main minion config file).
+#default_include: minion.d/*.conf
+
+# Set the location of the salt master server. If the master server cannot be
+# resolved, then the minion will fail to start.
+#master: salt
+master: saltmaster
```
Comment (renner) [inline docker/etc_minion1/salt/minion]: Same as above: can we minimize this and put it into `/etc/salt/minion.d`?

File: docker/etc_master/salt/master [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+#  - 'https://github.com/saltstack/salt-winrepo.git'
+
+### net-api configuration
+rest_cherrypy:
+  port: 8000
+  disable_ssl: True
+external_auth:
+  pam:
+    saltdev:
+      - .*
```
Comment (mbologna) [inline docker/etc_master/salt/master]: Yes, you're right on both the optimizations!

**Comment by mbologna** [comment]
Some refactoring has been done and now containers does not need any further configuration passed when launched.

This PR also contains a modified Travis build file to launch JUnit tests against these containers running salt-master and salt-api updated to the latest version of Salt shipped in the standard repository of the containers image (currently: OpenSUSE 42.2 with Salt 2016.3.4)

**Comment by lucidd** [comment]
@mbologna but what about tumbleweed? :stuck_out_tongue:

**Comment by mbologna** [comment]
@lucidd do you think it would be worth to rebase the Docker container using Tumbleweed or to create another one? ;)

**Comment by lucidd** [comment]
@mbologna haha no i was just kidding leap is fine :smile:

**Comment by mbologna** [comment]
@lucidd any chance we can merge this?

**Comment by lucidd** [comment]
@mbologna if the PR is updated to work with the current state state of the library sure.

**Comment by mbologna** [comment]
@lucidd PR updated and tests green!

**Pull request merged**